src/commands/lambda/flare.ts

```diff
@@ -1,14 +1,249 @@
+    }
+    if (this.credentials === undefined) {
+      this.context.stdout.write(commonRenderer.renderNoAWSCredentialsFound())
+      try {
+        await requestAWSCredentials()
+      } catch (err) {
+        this.context.stderr.write(commonRenderer.renderError(err))
+
+        return 1
+      }
+    }
+
+    // Get Lambda function configuration
+    this.context.stdout.write('\n🔍 Getting Lambda function configuration...\n')
+    const lambdaClientConfig: LambdaClientConfig = {
+      region,
+      credentials: this.credentials,
+    }
+    const lambdaClient = new LambdaClient(lambdaClientConfig)
+    let config
+    try {
+      config = await getLambdaFunctionConfig(lambdaClient, this.functionName ?? '')
+    } catch (err) {
+      this.context.stderr.write(
+        commonRenderer.renderError(`Unable to get Lambda function configuration: ${err.message}`)
+      )
+
+      return 1
+    }
+
+    // Print config
+    const configStrColored = util.inspect(config, false, undefined, true)
+    this.context.stdout.write(`\n${configStrColored}\n`)
+    if (this.isDryRun) {
+      this.context.stdout.write('\n🚫 The configuration was not sent as it was executed in dry run mode.\n')
+
+      return 0
+    }
+
+    // Send data to Datadog
+    const folderPath = path.join(process.cwd(), FLARE_OUTPUT_DIRECTORY)
+    const filePath = path.join(folderPath, FUNCTION_CONFIG_FILE_NAME)
+    const zipPath = path.join(folderPath, ZIP_FILE_NAME)
+    const configStrUncolored = JSON.stringify(config, undefined, 2)
+    try {
+      await writeFile(folderPath, filePath, configStrUncolored, this.context)
+      await zipContents(filePath, zipPath)
+      this.context.stdout.write('\n🚀 Sending to Datadog Support...\n')
+      await sendToDatadog(zipPath, this.caseId!, this.email!, this.apiKey!, this.context)
+      deleteFolder(folderPath, this.context)
+    } catch (err) {
+      this.context.stderr.write(commonRenderer.renderError(err.message))
+
+      return 1
+    }
+
+    return 0
+  }
+}
+
+/**
+ * Write the function config to a file
+ * @param folderPath
+ * @param filePath
+ * @param data
+ * @param context used for logging
+ * @throws Error if the file cannot be written
+ */
+export const writeFile = async (folderPath: string, filePath: string, data: string, context: BaseContext) => {
+  if (fs.existsSync(folderPath)) {
+    deleteFolder(folderPath, context)
+  }
+
+  try {
+    fs.mkdirSync(folderPath)
+    fs.writeFileSync(filePath, data)
+  } catch (err) {
+    throw Error(`Unable to create function configuration file: ${err.message}`)
+  }
+}
+
+/**
+ * Zip the contents of the flare folder
+ * @param filePath
+ * @param zipPath
+ * @throws Error if the zip fails
+ */
+export const zipContents = async (filePath: string, zipPath: string) => {
+  try {
+    const data = fs.readFileSync(filePath, 'utf8')
+    const zip = new JSZip()
+    zip.file(FUNCTION_CONFIG_FILE_NAME, data)
+    const content = await zip.generateAsync({type: 'nodebuffer'})
```

Comment:
couldn't we use `zlib` that is native? We wouldn't need a new dependency then.

Reply:
Thanks for the review, I will look into using `zlib` instead

Reply:
Hey @juan-fernandez 
Looks like `zlib` only supports zipping single files. While this would work for this PR, we do plan on zipping multiple files in the near future, so we would need a new dependency like `fszip`.

@duncanista What do you think? Should I use `zlib` for this PR or stick with `fszip`?

Reply:
@nhulston do you mean `jszip`? 

@juan-fernandez I'd normally go for the `zlib` approach too, when I researched on this, it seemed that we needed more packages to compress folders (`tar` and `fstream`). By using `zlib` and its dependencies for our use case, previously mentioned, we would be using 550~600 kB less than if we used `jszip`.

It seems that change of implementation might be trivial. But of course, this could be refactored in the future if package size is a problem. This would also not be merged directly into main since our working branch `serverless-flare` is still a WIP before its first release.

What do you think @juan-fernandez?